### PR TITLE
Refactored initial PoC code

### PR DESCRIPTION
**Pull request by BroiSatse**

### CODE REVIEW

File: app/lib/data_loader/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    def call
+      {
+        type: type,
+        current: current,
+        future: future,
+        increase: increase,
+        increase_percentage: increase_percentage
+      }
```
Comment (your) [inline app/lib/data_loader/calculator.rb:27]: you should be able to omit the hash value for all these keys

```
      {
        type:,
        current:,
        future:,
        increase:,
        increase_percentage:
      }
```

File: spec/models/pay_band_spec.rb [hunk 2]
```diff
@@ -0,0 +1,239 @@
+require 'rails_helper'
+
+RSpec.describe PayBand, type: :model do
+  # TODO: MOVE THIS LOGIC TO VALIDATOR OBJECT
```
Comment (your) [inline spec/models/pay_band_spec.rb:4]: This comment can now be removed

File: spec/models/pay_band_spec.rb [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    context 'with valid id' do
+      let(:id) { existing_pay_band.id }
+
+      it { is_expected.to be_a(described_class) }
+      it { is_expected.to have_attributes(id:) }
+    end
+
+  end
+
+  # describe '#spine_point?' do
```
Comment (your) [inline spec/models/pay_band_spec.rb:158]: This comment can now be removed

**Review by your — COMMENTED** [review]
Fantastic work, 🏅.
Just a few minor things and then we can merge it!

Also I've realised we need to add a workflow to trigger the tests when a PR is opened and not just when it's merged to main

File: app/lib/data_loader/calculator.rb (hunk 1 above)
Comment (BroiSatse) [reply app/lib/data_loader/calculator.rb:27]: True! Can't get used to that syntax. I wonder why they didn't just go for JS approach here `{type, current, ...}`

**Review by your — APPROVED** [review]
🚢

**Pull request merged**